chainer/functions/evaluation/binary_accuracy.py

```diff
@@ -0,0 +1,44 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class BinaryAccuracy(function.Function):
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        x_type, t_type = in_types
+
+        type_check.expect(
+            x_type.dtype == numpy.float32,
+            t_type.dtype == numpy.int32,
+            t_type.shape[0] == x_type.shape[0],
```

Comment:
As `x` and `t` are flatten below `x` and `t` can be more than one dimensional. So I think the requirement for the shapes is `prod(t_type.shape) == prod(x_type)` rather than the consistency of first dimensions.